spec/models/green_flag/feature_spec.rb

```diff
@@ -1,135 +1,187 @@
-      let(:site_visitor_id)  { 1 }
+    context 'the first applying rule is false' do
+      it 'returns a feature decision with the correct attributes' do
+        feature.stub(rules: [false_rule, true_rule])
 
-      let(:non_applying_rule) { double(id: 1, applies_to?: false, decision?: true)  }
-      let(:true_rule)         { double(id: 2, applies_to?: true,  decision?: true)  }
-      let(:false_rule)        { double(id: 3, applies_to?: true,  decision?: false) }
+        feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
 
-      context 'there are no rules' do
-        it 'returns a feature decision with the correct attributes' do
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+        expect(feature_decision.rule_id).to eq false_rule.id
+        expect(feature_decision.enabled).to eq false
+      end
+    end
+  end
 
-          expect(feature_decision.rule_id).to be_nil
-          expect(feature_decision.enabled).to be_nil
-        end
+  describe '#forget_non_manual_decisions!' do
+    let(:feature) { GreenFlag::Feature.create(code: 'big_font') }
+    let(:enabled) { true }
+
+    context 'when there are no decisions' do
+      subject { feature.forget_non_manual_decisions!(enabled) }
+
+      it 'should not attempt to delete non-manual decisions' do
+        expect(GreenFlag::FeatureDecision.non_manual).to receive(:delete_all).never
       end
 
-      context 'the first applying rule is true' do
-        it 'returns a feature decision with the correct attributes' do
-          feature.stub(rules: [non_applying_rule, true_rule])
+      it 'should not create a feature event' do
+        expect(GreenFlag::FeatureEvent.count).to eq 0
+      end
+    end
 
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+    context 'when there are manual decisions' do
+      subject { feature.forget_non_manual_decisions!(enabled) }
 
-          expect(feature_decision.rule_id).to eq true_rule.id
-          expect(feature_decision.enabled).to eq true
-        end
+      before(:each) do
+        @manual_fd = GreenFlag::FeatureDecision.create!(feature_id: feature.id, manual: true, site_visitor_id: 1)
+      end
+
+      it 'should not delete the manual decisions' do
+        expect(GreenFlag::FeatureDecision.find_by_id(@manual_fd.id)).to be_present
       end
 
-      context 'the first applying rule is false' do
-        it 'returns a feature decision with the correct attributes' do
-          feature.stub(rules: [false_rule, true_rule])
+      it 'should not create a feature event' do
+        expect(GreenFlag::FeatureEvent.count).to eq 0
+      end
+    end
 
-          feature_decision = feature.decide_if_enabled_for_visitor(site_visitor_id)
+    context 'when there are automatic decisions' do
+      before(:each) do
+        @feature2 = GreenFlag::Feature.create!(code: 'huge_banner_ad')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.